src/libImaging/Convert.c

```diff
@@ -1013,7 +1013,7 @@ p2l(UINT8 *out, const UINT8 *in, int xsize, const UINT8 *palette) {
     int x;
     /* FIXME: precalculate greyscale palette? */
     for (x = 0; x < xsize; x++) {
-        *out++ = L(&palette[in[x] * 4]) / 1000;
+        *out++ = round(L(&palette[in[x] * 4]) / 1000.0);
```

Comment:
Floating point arithmetic slower than integer. It's better to use `L24` macro here, it already contains rounding.

Reply:
Thanks. I've updated the commit.